isisd/isis_lsp.c

```diff
@@ -1070,6 +1071,80 @@ static struct isis_lsp *lsp_next_frag(uint8_t frag_num, struct isis_lsp *lsp0,
  * Builds the LSP data part. This func creates a new frag whenever
  * area->lsp_frag_threshold is exceeded.
  */
+void iteration_in_lsp_ip(struct isis_extended_ip_reach *r, struct isis_lsp *lsp,
```

Comment:
It is preferable to declare a function that return the count instead of void and returning the result with a variable. In particular, in the function, you don't check if the count variable has been allocated or not i.e. if it is not NULL.